### PR TITLE
♻️ Maintenance: Mypy `efs guardian` & `payments`

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- ♻️ Maintenance: Mypy `efs guardian` & `payments`

## Related issue/s
- https://github.com/ITISFoundation/osparc-simcore/issues/6115

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/payments/requirements/_base.txt
Comment (GitHK) [inline services/payments/requirements/_base.txt:1]: did you not upgrade a bit too much here?

File: services/payments/requirements/_test.in [hunk 1]
```diff
@@ -26,3 +26,7 @@ pytest-sugar
 python-dotenv
 python-socketio[asyncio_client]
 respx
+sqlalchemy[mypy]
+types-aiofiles # missing mypy stubs
+types-PyYAML # missing mypy stubs
+types-python-jose # missing mypy stubs
```
Comment (GitHK) [inline services/payments/requirements/_test.in:32]: just remove the comments not useful at all types and stubs it's clear what they are from the names of the libraries

File: services/payments/src/simcore_service_payments/services/auth.py [hunk 2]
```diff
@@ -64,7 +65,13 @@ def decode_access_token(token: str, settings: ApplicationSettings) -> str | None
...
 
 
-_credencial_401_unauthorized_exception_kwargs = {
+class _HTTPExceptionKwargs(TypedDict):
+    status_code: int
+    detail: str
+    headers: dict
+
+
+_credencial_401_unauthorized_exception_kwargs: _HTTPExceptionKwargs = {
```
Comment (GitHK) [inline services/payments/src/simcore_service_payments/services/auth.py:74]: `dict[str, Any]`also works

**Review by GitHK — APPROVED** [review]
Please check the dependencies, too many have been updated on the payments service

File: services/payments/src/simcore_service_payments/services/notifier_email.py [hunk 3]
```diff
@@ -206,11 +206,12 @@ async def _create_user_email(
...
-        _file_name = match.group("filename")
+        if match:
+            _file_name = match.group("filename")
 
-        attachment = MIMEApplication(pdf_response.content, Name=_file_name)
-        attachment["Content-Disposition"] = f"attachment; filename={_file_name}"
-        msg.attach(attachment)
+            attachment = MIMEApplication(pdf_response.content, Name=_file_name)
+            attachment["Content-Disposition"] = f"attachment; filename={_file_name}"
+            msg.attach(attachment)
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply services/payments/src/simcore_service_payments/services/notifier_email.py:214]: yes you are right I will add a log thanks.

File: services/payments/requirements/_base.txt
Comment (matusdrobuliak66) [reply services/payments/requirements/_base.txt:1]: no no i upgraded just the tests requirements

**Pull request merged**